File: Go/Algorithms/1071.go
package main

import "strings"

func gcdOfStrings(str1 string, str2 string) string {
	l, s := str1, str2
	if len(l) < len(s) {
		s, l = l, s
	}

	for ; len(s) != 0 && strings.HasPrefix(l, s); l, s = s, l {
		for ; strings.HasPrefix(l, s); l = l[len(s):] {
		}
	}
	if len(s) == 0 {
		return l
	}
	return ""
}

File: C/Algorithms/1071.c
char * gcdOfStrings(char * str1, char * str2){
    char *s = str1, *l = str2;
    if (strlen(s) > strlen(l)) {
        s = str2;
        l = str1;
    }
    
    while (*s && strncmp(l, s, strlen(s)) == 0) {
        int s_len = strlen(s);
        while (strncmp(l, s, s_len) == 0)
            l += s_len;
        char *temp = s;
        s = l;
        l = temp;
    }
    return *s == '\0' ? l : "";
}